src/jit/emitxarch.cpp

```diff
@@ -2903,14 +2903,27 @@ regNumber emitter::emitInsBinary(instruction ins, emitAttr attr, GenTree* dst, G
         varNum = tmpDsc->tdTempNum();
         offset = 0;
     }
+    else
+    {
+        // At this point we must have a memory operand that is a contained indir: if we do not, we should have handled
+        // this instruction above in the reg/imm or reg/reg case.
+        assert(mem != nullptr);
+        assert(memBase != nullptr);
+
+        if (memBase->OperGet() == GT_LCL_VAR_ADDR)
+        {
+            varNum = memBase->AsLclVarCommon()->GetLclNum();
+            offset = 0;
```

Comment:
This is a case of GT_IND(addr) and addr could have base, index, scale and offset.  Here we are making implicit assumption that if base= GT_LCL_VAR_ADDR then Index() is nullptr and scale and offset = 0.
I would suggest to explicitly assert the implicit assumptions here.